Objects/longobject.c

```diff
@@ -5260,6 +5260,31 @@ long_is_finite(PyObject *v)
 }
 #endif
 
+/*[clinic input]
+int.as_integer_ratio
+
+Return integer ratio.
+
+Return a pair of integers, whose ratio is exactly equal to the original int
+and with a positive denominator.
+
+>>> (10).as_integer_ratio()
```

Comment:
Do we need so much examples in a docstring?

Reply:
I think it's useful to have at least one positive and one negative example, so that it's obvious at a glance that the behaviour for negatives is to give (for example) `(-5, 1)` rather than `(5, -1)`. I could imagine users thinking that `0` was somehow a special case, too, so I like that we have the `0` example there.